### PR TITLE
Document fastcomp's bitcode linking advantage. Helps with #9817

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: site/source/docs/compiling/WebAssembly.rst [hunk 1]
```diff
@@ -65,6 +65,14 @@ upgrade from fastcomp to upstream:
     LLVM optimization passes by default, even if using LLVM IR in object files;
     for that you must pass ``--llvm-lto 1``.
 
+  * Another thing you might notice is that fastcomp's link stage runs LLVM
+    ``GlobalOpt``, which can do somethings that linking wasm object files can't,
```
Comment (dschuff) [inline site/source/docs/compiling/WebAssembly.rst]: I'm not sure describing this in terms of internal LLVM passes is really useful for users. (in other words, I guess I disagree with the statement that "you might notice" it :smile: ). What the user would notice is a performance regression, which might be mitigated using LTO.

Comment (aheejin) [reply site/source/docs/compiling/WebAssembly.rst]: Yeah, we can phrase this as something like 'fastcomp's link stage runs LTO in LLVM IR' or something.

Comment (sbc100) [reply site/source/docs/compiling/WebAssembly.rst]: How about "is able to perform some amount of link time optimization by default, where is the LLVM backend requires specific LTO flags".

Comment (dschuff) [reply site/source/docs/compiling/WebAssembly.rst]: Yeah that seems reasonable.

**Pull request merged**